File: ACT_4/py/main.py
import sys
from pathlib import Path
from antlr4 import *
from InterfacesLexer import InterfacesLexer
from InterfacesParser import InterfacesParser

def main():
    if len(sys.argv) < 2:
        print("Uso: python main.py <ruta_carpeta>")
        sys.exit(1)

    carpeta = Path(sys.argv[1])

    # Obtener todos los .txt y ordenarlos alfabéticamente
    archivos_txt = sorted(carpeta.glob('*.txt'))

    if not archivos_txt:
        print(f"No se encontraron archivos .txt en {carpeta}")
        return

    for archivo in archivos_txt:
        # 1. Leer archivo
        input_stream = FileStream(str(archivo), encoding='utf-8')

        # 2. Lexer
        lexer = InterfacesLexer(input_stream)
        stream_tokens = CommonTokenStream(lexer)

        # 3. Parser
        parser = InterfacesParser(stream_tokens)

        # 4. Construir el árbol (análisis)
        arbol = parser.configFile()

if __name__ == '__main__':
    main()

File: ACT_4/cpp/main.cpp
#include <iostream>
#include <fstream>
#include <filesystem>
#include <vector>
#include <algorithm>
#include <antlr4-runtime.h>
#include "InterfacesLexer.h"
#include "InterfacesParser.h"

using namespace antlr4;
namespace fs = std::filesystem;

int main(int argc, const char* argv[]) {
    if (argc < 2) {
        std::cerr << "Uso: ./parser_cpp <ruta_carpeta>" << std::endl;
        return 1;
    }

    std::string carpeta = argv[1];
    std::vector<std::string> archivos;

    // Obtener todos los archivos .txt del directorio
    for (const auto& entry : fs::directory_iterator(carpeta)) {
        if (entry.path().extension() == ".txt") {
            archivos.push_back(entry.path().string());
        }
    }

    // Ordenar alfabéticamente
    std::sort(archivos.begin(), archivos.end());

    // Procesar cada archivo
    for (const auto& ruta : archivos) {
        std::ifstream stream;
        stream.open(ruta);
        
        ANTLRInputStream input(stream);
        InterfacesLexer lexer(&input);
        CommonTokenStream tokens(&lexer);
        InterfacesParser parser(&tokens);
        
        tree::ParseTree *arbol = parser.configFile();
    }

    return 0;
}
